### PR TITLE
Add `saveResponse` option

### PR DESCRIPTION
**Pull request by PsychoXIVI**
Solves #8, using [@mikeecb solution](https://github.com/sitespeedio/chrome-har/pull/41#issuecomment-456913596)

btw, Chrome can't parse well generated HAR files (not our fault). Here is [the issue on Chromium](https://bugs.chromium.org/p/chromium/issues/detail?id=968623).

Still you can use [softwareishard.com HAR Viewer](http://www.softwareishard.com/har/viewer/)

Edit: Multiple force-pushes due to not properly tested code before commiting :|

### CODE REVIEW

**Comment by mickeyreiss** [comment]
I'm using this and it appears to be _mostly_ working! However I suspect there is a race condition where stop does not block on all requests loading in. Is it possible that there's a race condition around starting a request and waiting for the response data to be read?

**Comment by PsychoXIVI** [comment]
I think you might be right. There should be a flag or something set at `stop`  to prevent observing more and more events while `Promise.all(this.response_body_promises)` is being waiting for. It isn't very feature breaking change, but I some crafted cases could make the har collector unusable at all (continuously loading big responses). 

I will try to fix it in commit in few moments.

**Comment by mickeyreiss** [comment]
Thanks for the update. I'll run this on Monday to see if this resolves my race condition.

File: lib/PuppeteerHar.js [hunk 1]
```diff
@@ -29,27 +29,73 @@ class PuppeteerHar {
...
             this.client.on(method, params => {
+                if (!inProgress) {
+                    return;
+                }
                 this.page_events.push({ method, params });
             });
         });
         network_observe.forEach(method => {
             this.client.on(method, params => {
+                if (!inProgress) {
```
Comment (Everettss) [inline lib/PuppeteerHar.js]: isn't should be `this.inProgerss` here?

File: lib/PuppeteerHar.js [hunk 2]
```diff
@@ -29,27 +29,73 @@ class PuppeteerHar {
...
+    async start({ path, saveResponse } = {}) {
+        this.inProgress = true;
+        this.saveResponse = saveResponse || false;
         this.path = path;
         this.client = await this.page.target().createCDPSession();
         await this.client.send('Page.enable');
         await this.client.send('Network.enable');
         page_observe.forEach(method => {
             this.client.on(method, params => {
+                if (!inProgress) {
```
Comment (Everettss) [inline lib/PuppeteerHar.js]: isn't should be `this.inProgerss` here?

File: package.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
   "author": "Michał Janaszek",
   "license": "MIT",
   "dependencies": {
-    "chrome-har": "^0.4.1"
+    "chrome-har": ">=0.10.0"
```
Comment (Everettss) [inline package.json]: please use `^0.10.0` notation

**Pull request merged**